service/src/main/java/com/derpgroup/quip/bots/insultibot/resource/InsultiBotAlexaResource.java

```diff
@@ -130,10 +145,10 @@ public SpeechletResponseEnvelope doAlexaRequest(SpeechletRequestEnvelope request
       Reprompt reprompt = null;
       SsmlOutputSpeech outputSpeech;
       
-      switch(voiceInput.getMessageType()){
-      case END_OF_CONVERSATION:
-      case STOP:
-      case CANCEL:
+      switch(serviceInput.getSubject()){
```

Comment:
This was all built before we had the serviceOutput object with the flag on it.  We could probably handle this in the manager now...